mixer/pkg/il/compiler/compiler_test.go

```diff
@@ -792,6 +806,13 @@ end`,
 	{
 		expr: `ar["c"] | "foo"`,
 		input: map[string]interface{}{
+			"ar": map[string]string{},
+		},
+		result: "foo",
+	},
+	{
+		expr: `ar["c"] | "foo"`,
+		input: map[string]interface{}{
```

Comment:
Consider adding test for below cases where the attribute reference (source.absentInBag) does not exist . May be it already exists.

attribute reference | ""
attribute reference | "foo"

Reply:
There are tests for the second case already.
https://github.com/istio/istio/blob/master/mixer/pkg/il/compiler/compiler_test.go#L414
The first case simply returns "" as expected. I'll add tests in my follow-up test-refactoring pack.